### PR TITLE
Add PenaltyWeighting

### PR DESCRIPTION
**Pull request by hcourt**
Instead of dividing the edge time (milliseconds) by a priority, multiply the time by a penalty. This way is easier to reason about.

Tested by routing and clicking around to make sure the routes still look sane.

### CODE REVIEW

File: core/src/main/java/com/graphhopper/routing/weighting/PenaltyWeighting.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public class PenaltyWeighting extends FastestWeighting {
+
+    private final double minFactor;
+    private final double maxPenalty;
+    private final DecimalEncodedValue penaltyEnc;
+
+    public PenaltyWeighting(FlagEncoder encoder, PMap pMap, TurnCostProvider turnCostProvider) {
+        super(encoder, pMap, turnCostProvider);
+        penaltyEnc = encoder.getDecimalEncodedValue(EncodingManager.getKey(encoder, "penalty"));
+        minFactor = 1 / PenaltyCode.getValue(EXCLUDE.getValue());
```
Comment (abhumbla) [inline core/src/main/java/com/graphhopper/routing/weighting/PenaltyWeighting.java:43]: change to 1

File: core/src/main/java/com/graphhopper/routing/util/PenaltyCode.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    PenaltyCode(int value) {
+        this.value = value;
+    }
+
+    public int getValue() {
+        return value;
+    }
+
+    public static double getFactor(int value) {
+        return (double) value / 10.0;
```
Comment (abhumbla) [inline core/src/main/java/com/graphhopper/routing/util/PenaltyCode.java:51]: change to 1?

**Review by abhumbla — APPROVED** [review]
LGTM

**Review by graue — APPROVED** [review]
Looks good, excited to see how this can lead to improved routing!

**Pull request merged**